### ISSUE TITLE
Server sync - can't add certificate

### ISSUE THREAD

**Issue by LeandreNguyen** [MISP/MISP#7544]
Hello,

I'm trying to set up a synchronization with another MISP instance using TLS. Problem is, I can't seem to upload the .pem certificate when adding the server : nothing happens when I select a file and no error messages are raised. 
I tried going through the OpenAPI, but I don't quite understand what's expected for the fields "cert_file" or "client_cert_file".

Any advice ?

**Comment by mokaddem** [comment]
Hello,
Could you make sure that you are running the latest version of MISP?
Also, would you mind describing the exact procedure you use when trying to add a certificate?
Thanks!

**Comment by LeandreNguyen** [comment]
I checked, and I'm indeed running 2.4.146, so that's probably not the issue.
As for the procedure, I'm connecting to the /servers/edit/<ID> page and clicking the "Add certificate file" under either Server certificate or Client certificate. That opens up the file explorer, which lets me select a certificate in .pem format. The file explorer closes, but the certificate isn't uploaded (still shows "Not set"). No logs or errors, as far as I can tell.

I'm leaning towards a JS (or maybe redis ?) issue, probably caused by my implementation since it doesn't seem to be a common problem. No idea how to solve it, though. I've considered going through the API or the database directly, but I'm not sure how.

**Comment by mokaddem** [comment]
I just tried the exact same procedure and indeed there is no feedback from the UI that a certificate has been selected.
However, I was still able to save it and then it gets displayed and used correctly on that same page and on the servers index.
Could you confirm that you are only experiencing bad UI feedback?
Thanks!

**Comment by maverickvn360** [comment]
Same experience. But I think it's not a major cause for concern. Replacing a server/client cert directly, in my case, seems to have no effect for the first time - and is added when selected the second time. Maybe an error in server edit files..

**Comment by LeandreNguyen** [comment]
I wish I could, but no. The problem is that selecting a certificate has no effect at all : it isn't saved, and doesn't get used or displayed on any page afterwards.

**Comment by maverickvn360** [comment]
Could you please share a screenshot after selecting the certificate and saving the server?

So - as @mokaddem said, when you do save the server config, you can see the server certs filled as 'X.pem' and 'X_client.pem', where 'X' is the server ID (of sorts).

As for the certificates, the public cert of the corresponding server goes to 'Server Certificate file' and the private key/cert of this server (which is signed with the server/CA key) goes in 'Client Certificate file'.

If it's not saved, just try doing it again, and you might as well see the fields populated while the server is indexed.

**Comment by LeandreNguyen** [comment]
Sure, here's a screenshot of the servers index after trying to save the certificates. As you can see, the relevant fields are empty.

![Capture_2](https://user-images.githubusercontent.com/85225326/126279214-a12b04c8-39b3-40be-9542-41c0b76d25ea.PNG)

Thanks for the information, that's good to know. On that subject, is there any restriction on the filename or the file itself beyond the .pem extension ?

And yes, I did try several times under various conditions, but I never had much success.

**Comment by maverickvn360** [comment]
Yes, no capital letters allowed and the name should start with a-z/0-9/_/[dot]. For the extension though, I've tried uploading crt and cer files, seems to have no effect whatsoever. Skimming the code, there might be restrictions on the size too, 5MiB.

Apparently, the files are copied into the

`/var/www/MISP/app/files/certs/`

folder. It could be worth trying to copy the cert files under this folder (server cert -> 2.pem and client cert -> 2_client.pem). And you did mention there are no logs (under the logs tab or the error folder). Could you please check again and if there are none present, there might be an implementation issue that I'm not aware of.

**Comment by simillo01** [comment]
@maverickvn360 solution works on MISP 2.5 on Ubuntu 24.04, because for me GUI it's bugged so i cannot upload the client certificate properly.
